src/backend/distributed/utils/ruleutils_95.c

```diff
@@ -3624,7 +3624,7 @@ get_variable(Var *var, int levelsup, bool istoplevel, deparse_context *context)
 	else
 	{
 		/* System column - name is fixed, get it from the catalog */
-		attname = get_rte_attribute_name(rte, attnum);
+		attname = get_relid_attribute_name(rte->relid, attnum);
```

Comment:
This was changed because otherwise:

```sql
select tableoid, xmin, xmax from wikipedia_changes limit 100;
ERROR:  invalid attnum -7 for rangetable entry wikipedia_changes
```

I'm not sure if this is an oversight in Postgres or if it's indicitive of a problem with how Citus uses `get_variable()`, but with the code as it is, it's not possible to select system columns from worker shards without this change.